Content.Shared/Devour/Components/DevourerComponent.cs

```diff
@@ -0,0 +1,82 @@
+using Content.Shared.Actions;
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.Chemistry.Reagent;
+using Content.Shared.Devour;
+using Content.Shared.Whitelist;
+using Robust.Shared.Audio;
+using Robust.Shared.Containers;
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Devour.Components;
+
+[RegisterComponent, NetworkedComponent]
+[Access(typeof(SharedDevourSystem))]
+public sealed class DevourerComponent : Component
+{
+    [DataField("devourActionId", customTypeSerializer: typeof(PrototypeIdSerializer<EntityTargetActionPrototype>))]
+    public string DevourActionId = "Devour";
+
+    [DataField("devourAction")]
+    public EntityTargetAction? DevourAction;
+
+    [ViewVariables(VVAccess.ReadWrite), DataField("soundDevour")]
+    public SoundSpecifier? SoundDevour = new SoundPathSpecifier("/Audio/Effects/demon_consume.ogg")
+    {
+        Params = AudioParams.Default.WithVolume(-3f),
+    };
+
+    [DataField("devourTime")]
+    public float DevourTime = 3f;
+
+    /// <summary>
+    /// The amount of time it takes to devour something
+    /// <remarks>
+    /// NOTE: original intended design was to increase this proportionally with damage thresholds, but those proved quite difficult to get consistently. right now it devours the structure at a fixed timer.
+    /// </remarks>
+    /// </summary>
+    [DataField("structureDevourTime")]
+    public float StructureDevourTime = 10f;
+
+    [ViewVariables(VVAccess.ReadWrite), DataField("soundStructureDevour")]
+    public SoundSpecifier? SoundStructureDevour = new SoundPathSpecifier("/Audio/Machines/airlock_creaking.ogg")
+    {
+        Params = AudioParams.Default.WithVolume(-3f),
+    };
+
+    /// <summary>
+    /// Where the entities go when it devours them, empties when it is butchered.
+    /// </summary>
+    public Container Stomach = default!;
```

Comment:
Stomachs are already a thing. Instead of having 2 different types of stomachs for devoured food and non-devoured food, regular stomachs should be able to store entities instead of just reagents.

Reply:
StomachComponent don't have a container for now, it can store only reagents. I think your suggestion is a right thing to do, but I don't want to bloat the scope of this PR and also change StomachSystem\Components.